### PR TITLE
Change metadata for plugin ref docs and deprecation doc

### PR DESCRIPTION
**Pull request by mdlinville**
**- What I did**

Updated metadata of plugin docs and deprecation doc from TOML to YAML
Synced metadata `title` to H1 headers, removed H1 headers

Follow-up fro #28174 

@thaJeztah @dnephin @johndmulhausen PTAL

### CODE REVIEW

File: docs/extend/manifest.md [hunk 1]
```diff
@@ -1,19 +1,11 @@
...
-[menu.main]
-parent = "engine_extend"
-weight=1
-+++
-<![end-metadata]-->
-
-# Plugin Manifest Version 0 of Plugin V2
+---
+redirect_from:
+- "/engine/extend/"
```
Comment (thaJeztah) [inline docs/extend/manifest.md]: I think this was a copy-paste error, and should be removed

Comment (thaJeztah) [inline docs/extend/manifest.md]: (in the original document)

**Review by thaJeztah — CHANGES_REQUESTED** [review]
found one issue, but LGTM otherwise

**Comment by mstanleyjones** [comment]
Fixed. PTAL again.

**Review by thaJeztah — APPROVED** [review]
LGTM!

**Pull request merged**